src/tools/select.rs

```diff
@@ -136,23 +136,44 @@ impl MouseDelegate<EditSession> for Select {
 
     fn left_down(&mut self, event: &MouseEvent, data: &mut EditSession) {
         if event.count == 1 {
-            let sel = data.iter_items_near_point(event.pos, None).next();
+            let sel = data.hit_test_all(event.pos, None);
             if let Some(point_id) = sel {
                 if !event.mods.shift() {
                     // when clicking a point, if it is not selected we set it as the selection,
                     // otherwise we keep the selection intact for a drag.
                     if !data.selection.contains(&point_id) {
-                        data.selection_mut().clear();
-                        data.selection_mut().insert(point_id);
+                        data.set_selection_one(point_id);
                     }
                 } else if !data.selection_mut().remove(&point_id) {
                     data.selection_mut().insert(point_id);
                 }
+            } else if let Some((seg, _t)) = data.hit_test_segments(event.pos, None) {
+                // TODO: make these non-draggable.
```

Comment:
Can you clarify this? My expectation is that I should be able to alt+click on the line segment, and then drag to modify the curve, although we don't have manipulation of segments (vs elements) at the moment..

Reply:
Sure. Right now the behavior is pretty bad, you get the click behavior, but it also creates a rectangular drag. Funny enough, that's also what Glyphs does for alt-click. For regular clicking on segments, Glyphs only does the segment selection behavior on mouse-up, and then only if the gesture did not become a drag. I'm not actually convinced this is worth emulating precisely; it seems a bit inconsistent. I personally have this filed under "further research and experimentation".

I'm personally not a huge fan of the drag curve segment to modify it, because I feel it doesn't give enough control (so I basically never use it), but do see the value in having that be the behavior for alt-click + drag on a line segment.

For the time being, I'll probably suppress the drag behavior, the same way I do in the pen tool, and add drag behavior later when I have more clarity exactly how it should work.